llvm/include/llvm/RemoteCachingService/Client.h

```diff
@@ -87,12 +88,19 @@ class KeyValueDBClient {
       --NumPending;
       return receiveNextImpl();
     }
+    Expected<Response> getValueSync(std::string Key) {
```

Comment:
The new `*Sync` functions seem out-of-place when put inside the `*AsyncQueue` classes, how about moving them to the top classes, `KeyValueDBClient` and `CASDBClient`?

Reply:
The stub is owned by queue currently but we can change that for sure. There should be more code sharing as well, which I am planning to improve. Let me give it another thought.

Reply:
AFAICT the stub is owned by `CASDBClientImpl` and `KeyValueDBClientImpl`, the queues get references to the stub.